### PR TITLE
chore(chart): Save modal select placeholder value

### PR DESCRIPTION
**Pull request by nikolagigic**
### SUMMARY
Fix SaveModal dashboard select placeholder value and clear button.
https://github.com/apache/superset/issues/12366

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:
104073894-b578d680-51c3-11eb-81c3-d2714db4bef6
After:
Screenshot 2021-01-11 at 19 02 48

### TEST PLAN
1. **Open** or **create** any chart
2. Click on **Save** button to save the chart

**Expected result**: Under **ADD TO DASHBOARD** label, select dropdown should have a placeholder text with no clear button. If the value is selected, the value should be displayed instead of placeholder and clear button should appear.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by nikolagigic** [comment]
@junlincc @rusackas

**Comment by nikolagigic** [comment]
🎫  viz:explore:ui

File: superset-frontend/src/explore/components/SaveModal.tsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ import { connect } from 'react-redux';
 
 // Session storage key for recent dashboard
 const SK_DASHBOARD_ID = 'save_chart_recent_dashboard';
-const SELECT_PLACEHOLDER = t('**Select** a dashboard OR **create** a new one');
+const SELECT_PLACEHOLDER = t('Select dashboard');
```
Comment (ktmud) [inline superset-frontend/src/explore/components/SaveModal.tsx:32]: Can we keep the original copy as I don't think we've added the functionality to create a dashboard in the dropdown yet. 

image

https://projects.invisionapp.com/share/V5VH03AHBER#/screens/399817652

cc @mihir174

For context, #10355 added this copy and there is a discussion on why this is still not optimal.

Comment (nikolagigic) [reply superset-frontend/src/explore/components/SaveModal.tsx:32]: @ktmud but the original text is referring to dashboard creation through save modal?

Comment (ktmud) [reply superset-frontend/src/explore/components/SaveModal.tsx:32]: Have we removed that functionality?

Comment (adam-stasiak) [reply superset-frontend/src/explore/components/SaveModal.tsx:32]: It is possible to create a dashboard through this modal and I guess text should claim that creation is possible.

**Comment by adam-stasiak** [comment]
I have a behavior where when I add new chart and try to save it is automatically selected first dashboard from list:


https://user-images.githubusercontent.com/25153919/104231907-e3724b00-544f-11eb-9e4b-7ccc1d1c6c0b.mov


I guess it should not be chosen by default to avoid mistakes. In previous releases it was not proposed.
image

**Comment by ktmud** [comment]
> I guess it should not be chosen by default to avoid mistakes. In previous releases it was not proposed.

When users come from "View chart in Explore" on the dashboard page, the origin dashboard should be selected (so they can click on "Save & Go to Dashboard").

When users edit an existing chart from the Chart List page, the dashboard should not be selected.

**Comment by mihir174** [comment]
thanks for the context @ktmud 

Here's a common UI pattern I've seen in many products for this scenario - 

Slice 1 (5)

**Comment by nikolagigic** [comment]
Dashboards should now be selected properly and I changed the placeholder text

**Review by junlincc — APPROVED** [review]
LGTM! thanks for the PR! @villebro @zhaoyongjie ready for code review. @ktmud take another look? 

Explore save modal deserves a much bigger redesign. I can easily name a handful of UX issues.😞 Nothing critical, we can wait. 

ezgif-7-3ca75cef6b86

**Review by rusackas — APPROVED** [review]
LGTM! Thanks to everyone for the input! I appreciate all the consideration (and empathy) that goes into these details!

**Pull request merged**